### PR TITLE
Fibonacci works

### PR DESCRIPTION
**Pull request by SzymonGajewski**

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+    int F1 = 1;
+    
+    if(sequence == 1 || sequence == 0){
+        Fnext = sequence;
+    }
+    else{
+        for(int i = 2; i <= sequence; ++i){        
+            Fnext = F0 + F1;
+            F0 = F1;
+            F1 = Fnext;
```
Comment (piezol) [inline module1/homework/fibonacci/fibonacci.hpp:15]: That loop does not have to be inside 'else'.

File: module1/homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence == 1 || sequence == 0){
+        return sequence;
+    }
+    else{
+        return fibonacci_recursive(sequence - 1) + fibonacci_recursive(sequence - 2);
+    }
```
Comment (DamiXioR) [inline module1/homework/fibonacci/fibonacci.hpp:27]: i think you can skip last else and just use return

File: module1/homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,28 @@
 #pragma once
 
-int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+int fibonacci_iterative(int sequence){
+    int Fnext{};
+    int F0{};
```
Comment (DamiXioR) [inline module1/homework/fibonacci/fibonacci.hpp:5]: in my opinion the better practice is to initialize variables with 0

Comment (SzymonGajewski) [reply module1/homework/fibonacci/fibonacci.hpp:5]: I thought that {} for int is same as = 0, isn't it? Or You mean just how looks better?

Comment (ziobron) [reply module1/homework/fibonacci/fibonacci.hpp:5]: It is the same and you can use them interchangeably. Which looks better is a personal matter :)

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1430], "points": [2, 6] }

**Pull request closed without merging**